cpp/serve/engine_actions/eagle_new_request_prefill.cc

```diff
@@ -115,6 +121,9 @@ class EagleNewRequestPrefillActionObj : public BatchPrefillBaseActionObj {
         // Speculative models shift left the input tokens by 1 when base model has committed tokens.
         // Note: for n > 1 cases Eagle doesn't work because parent entry doesn't shift input tokens.
         for (int j = 0; j < static_cast<int>(input_data.size()); ++j) {
+          if (!model_id) {
```

Comment:
model_id != 0 ?

Reply:
actually, here does not matter if model == 0. And prefix cache only cares about. base model(model_id == 0), so we only update if model_id == 0. However, it will be changed if input data logic refactored soon, by using offset instead.